### PR TITLE
[Uid] improve the API of Uuid

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

### CODE REVIEW

File: src/Symfony/Component/Uid/Uuid.php [hunk 1]
```diff
@@ -100,27 +95,23 @@ public function getType(): int
         return uuid_type($this->uuid);
     }
 
-    public function getVariant(): int
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Uid/Uuid.php]: Nobody uses variants, isn't it? Let's drop it to make our API simpler.
`uuid_variant()` still exists for ppl that need it.

File: src/Symfony/Component/Uid/Uuid.php [hunk 2]
```diff
@@ -100,27 +95,23 @@ public function getType(): int
...
-    {
-        return uuid_variant($this->uuid);
-    }
-
-    public function getTime(): int
+    public function getTime(): float
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Uid/Uuid.php]: I don't get why `uuid_time` is limited to an integer timestamp while the UUID time is a 60bit number. Let's return a float instead, with subsecond info.

File: src/Symfony/Component/Uid/Uuid.php [hunk 3]
```diff
@@ -100,27 +95,23 @@ public function getType(): int
...
     {
         if (self::TYPE_1 !== $t = uuid_type($this->uuid)) {
             throw new \LogicException("UUID of type $t doesn't contain a time.");
         }
 
-        return uuid_time($this->uuid);
-    }
+        $time = '0'.substr($this->uuid, 15, 3).substr($this->uuid, 9, 4).substr($this->uuid, 0, 8);
 
-    public function getMac(): string
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Uid/Uuid.php]: Same as getVariant: nobody needs this IMHO - ppl that do can still use `uuid_mac()`

**Comment by nicolas-grekas** [comment]
Tests need https://github.com/symfony/polyfill/pull/241 to be green.

**Comment by lyrixx** [comment]
@nicolas-grekas I guess you missed click. You closed the PR

**Comment by nicolas-grekas** [comment]
Actually not, I'd like to split the PR in chunks (and provide a maybe different end result :))
Check #36053 for the 2st step

**Pull request closed without merging**